client/client.go

```diff
@@ -21,8 +21,6 @@ type Client struct {
 	Password       *string
 	APIEndpoint    *url.URL
 	httpClient     *http.Client
-	authCookie     *http.Cookie
-	cookieIssued   time.Time
```

Comment:
I removed the platform specific auth methods from the top level client and put them into their respective clients. Compute uses cookies and Storage uses headers